New guard for apache mod_auth_openidc authentication

sgomez opened the pull request:
This PR adds a new guard to support authentication from `mod_auth_openidc` and add future required logic like redirect to custom user domain. This guard does not support role configuration with user attributes as `mod_auth_mellon` (#2) does.

Only one auth method will be available and can be configured with APP_MOD_SECURITY environment variable. Mellon authentication is configured by default.

No BC expected.

Pending:

- [x] Update README

thijskh commented on config/packages/security.yaml:
```diff
@@ -22,8 +22,8 @@ security:
-        - { path: '^/connect/(mellon)$', role: 'IS_AUTHENTICATED_ANONYMOUSLY' }
-        - { path: '^/connect/(mellon)/check', role: 'ROLE_USER' }
+        - { path: '^/connect/(mellon|inacademia)$', role: 'IS_AUTHENTICATED_ANONYMOUSLY' }
+        - { path: '^/connect/(mellon|inacademia)/', role: 'ROLE_USER' }
```
So it's only specifically usable with inacademia and not with a generic OpenID Connect provider?

sgomez replied:
No, the point is I asked for this return url to @surfnet-niels because I thought it is more semantic for our use case. Now, I think I was wrong and this should be 'oidc'. This wil require another change in the return url for our demo.

sgomez replied:
Changed to oidc

thijskh commented on README.md:
```diff
@@ -48,88 +48,130 @@ parameters:
 
+### With OpenID Connect Authentication
+
+OpenID Connect authentication use Apache [openidc authentication module](https://github.com/zmartzone/mod_auth_openidc).
```
"uses the"

thijskh reviewed (commented):
How is logout handled?

sgomez commented:
@thijskh logout is handled removing the session cache in Symfony as any other application. In SAML authentication we can do logout in the IdP but this is not possible in OIDC.

The user could be still logged in the OP, but the oidc module can be configured with the prompt parameter to force user to sign in every time.

thijskh commented:
Alright. I wondered because `app.security.logouturl` is still set to the mellon location.

sgomez commented:
@thijskh you were right, I wasn't considering apache_mod_openidc cookies. So each module have a different url. I added a new parameter to configure it (APP_MOD_LOGOUT). By default is the mellon logout url, so this shouldn't break anything in your environment.

To configure the login button label I added a APP_IDP_NAME parameter too (with SURFconext value by default).
